projects/button_adc/src/main.c

```diff
@@ -0,0 +1,59 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+static void prv_adc_callback(const GpioAddress *address, void *context) {
+  AdcChannel *analogChannel = context;
+  uint16_t analogPinData = 0;
+  adc_read_raw(*analogChannel, &analogPinData);
+  LOG_DEBUG("Analog Pin data: %d\n", analogPinData);
+}
+
+int main(void) {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAddress analogPin = { .port = GPIO_PORT_A, .pin = 6 };
+  GpioAddress button = { .port = GPIO_PORT_B, .pin = 2 };
+
+  GpioSettings pinSettings = {
+    GPIO_DIR_IN,        //
+    GPIO_STATE_LOW,     //
+    GPIO_RES_NONE,      //
+    GPIO_ALTFN_ANALOG,  //

```

Comment:
Please use the GNU-style `.direction =`, `.state =`, etc when initializing settings structs like this. It makes it much easier for humans to parse and lets you move around the lines without changing the meaning. This also goes for the `InterruptSettings` struct a couple lines down.